src/plugins/vis_type_table/public/plugin.ts

```diff
@@ -30,6 +30,16 @@ export interface TableVisPluginStartDependencies {
   data: DataPublicPluginStart;
 }
 
+const setupTableVis = async (
+  core: CoreSetup,
+  { expressions, visualizations }: TableVisPluginSetupDependencies
+) => {
+  const [coreStart] = await core.getStartServices();
+  expressions.registerFunction(createTableVisFn);
+  expressions.registerRenderer(tableVisRenderer(coreStart));
```

Comment:
This pattern is used in other places although the confusion i think is in the function name..e.g. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d7004dc5b0392477fdd54ac66b29d231975a173b/src/plugins/vis_type_tagcloud/public/plugin.ts#L81

Changing the name from `tableVisRenderer` to `getTableVisRenderer` should make it clearer and is the naming convention followed in other places. and since this is only run during setup to bootstrap the context provider, i think it is safe to pass all the core start services.

Reply:
will change to `getTableVisRenderer`

Reply:
for `coreStart`, I am using something from core in upcoming format PR. I will see if I need it.